nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Updater/WebApplicationTargetsConditionPatcher.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+
+using Microsoft.Language.Xml;
+
+namespace NuGetUpdater.Core.Updater
+{
+    internal class WebApplicationTargetsConditionPatcher : IDisposable
+    {
+        private string? _capturedCondition;
+        private readonly XmlFilePreAndPostProcessor _processor;
+
+        public WebApplicationTargetsConditionPatcher(string projectFilePath)
+        {
+            _processor = new XmlFilePreAndPostProcessor(
+                getContent: () => File.ReadAllText(projectFilePath),
+                setContent: s => File.WriteAllText(projectFilePath, s),
+                nodeFinder: doc => doc.Descendants()
+                    .FirstOrDefault(e => e.Name == "Import" && e.GetAttributeValue("Project") == @"$(VSToolsPath)\WebApplications\Microsoft.WebApplication.targets")
+                    as XmlNodeSyntax,
+                preProcessor: n =>
+                {
+                    var element = (IXmlElementSyntax)n;
+                    _capturedCondition = element.GetAttributeValue("Condition");
```

Comment:
If you DO make the above return multiple nodes that makes this assignment much harder to get correct, so maybe it's a bad idea? Just spit-balling, follow your heart.